.github/workflows/ContinuousIntegration.yml

```diff
@@ -0,0 +1,36 @@
+# GitHub workflow to enable continuous integration
+name: Continuous Integration
+
+# This workflow is triggered on pushes and pull requests to the repository.
+on: 
+  push:
+    branches: '**'
+  pull_request: 
+    branches: 'master'
+
+jobs:
+  build:
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ ubuntu-18.04, macos-10.15 ]
+        cxx: [ g++-9, clang++ ]
+        build_type: [ Debug, Release ]
+
+    steps:
+    - name: which CXX
+      run: |
+        which ${{matrix.cxx}} 
+        ${{matrix.cxx}} --version
+    - uses: actions/checkout@v2
+    - name: mkdir bin
+      run: mkdir bin 
+    - name: cmake
+      run: cmake -D CMAKE_CXX_COMPILER=`which ${{matrix.cxx}}` -D CMAKE_BUILD_TYPE=${{matrix.build_type}} ..  
+      working-directory: ./bin
+    - name: make
+      run: make
```

Comment:
You should have copied the one used for interpolation. It has the -j2 on the make command. It's a minor thing.